src/services/datamigration/datamigrationService.js

```diff
@@ -0,0 +1,225 @@
+const Finsemble = require("@chartiq/finsemble");
+const BaseService = Finsemble.baseService;
+const {
+	RouterClient,
+	Logger,
+	StorageClient,
+	LauncherClient
+} = Finsemble.Clients;
+
+Logger.start();
+
+/**
+ * class DataMigrationService
+ * 
+ * This class is configurable to migrate data from one storage adapter to another.
+ * In this example, IndexedDB data is being migrated to LocalStorage for all internal Finsemble data.
+ * 
+ */
+class DataMigrationService extends BaseService {
+	constructor(args) {
+		super(args);
+		this.adapter = "IndexedDBAdapter";
+		this.newAdapter = "LocalStorageAdapter";
+	}
+
+	/**
+	 * retrieveDatabase
+	 * 
+	 * Takes all Storage Adapter topics provided and creates an array of objects for all topics
+	 * Invokes saveAllData afterwards.
+	 * 
+	 * @param {Array} topics The storage topics to be migrated as specified in TOPICS
+	 */
+	retrieveDatabase(topics) {
+		const promiseSet = [];
+
+		topics.forEach((topic) => {
+			promiseSet.push(new Promise((resolve, reject) => {
+				// Reset to use old storage adapter to retrieve all the user's data
+				StorageClient.setStore({ topic: topic, dataStore: this.adapter }, (err, data) => {
+					if (err) {
+						reject(new Error(err));
+					}
+					resolve(data);
+				});
+			}));
+		});
+
+		Promise.all(promiseSet).then((data) => {
```

Comment:
You have `reject` statements in your promises, but you don't have a `catch` clause to catch those rejections.